src/url_search_params.rs

```diff
@@ -0,0 +1,352 @@
+    /// assert_eq!(params.get("a"), Some("1"));
+    /// assert_eq!(params.get("b"), Some("2"));
+    /// ```
+    pub fn parse(input: &str) -> Self {
+        Self(unsafe { ffi::ada_parse_search_params(input.as_ptr().cast(), input.len()) })
+    }
+
+    /// Returns the size of the URLSearchParams struct.
+    ///
+    /// ```
+    /// use ada_url::URLSearchParams;
+    /// let params = URLSearchParams::parse("a=1&b=2");
+    /// assert_eq!(params.size(), 2);
+    /// ```
+    pub fn size(&self) -> usize {
+        unsafe { ffi::ada_search_params_size(self.0) }
+    }
+
+    /// Sorts the keys of the URLSearchParams struct.
+    pub fn sort(&self) {
+        unsafe { ffi::ada_search_params_sort(self.0) }
+    }
+
+    /// Appends a key/value to the URLSearchParams struct.
+    pub fn append(&self, key: &str, value: &str) {
+        unsafe {
+            ffi::ada_search_params_append(
+                self.0,
+                key.as_ptr().cast(),
+                key.len(),
+                value.as_ptr().cast(),
+                value.len(),
+            )
+        }
+    }
+
+    /// Removes all keys pre-existing keys from the URLSearchParams struct
+    /// and appends the new key/value.
+    ///
+    /// ```
+    /// use ada_url::URLSearchParams;
+    /// let params = URLSearchParams::parse("a=1&b=2");
+    /// params.set("a", "3");
+    /// assert_eq!(params.get("a"), Some("3"));
+    /// ```
+    pub fn set(&self, key: &str, value: &str) {
+        unsafe {
+            ffi::ada_search_params_set(
+                self.0,
+                key.as_ptr().cast(),
+                key.len(),
+                value.as_ptr().cast(),
+                value.len(),
+            )
+        }
+    }
+
+    /// Removes a key/value from the URLSearchParams struct.
+    /// Depending on the value parameter, it will either remove
+    /// the key/value pair or just the key.
+    ///
+    /// ```
+    /// use ada_url::URLSearchParams;
+    /// let params = URLSearchParams::parse("a=1&b=2");
+    /// params.remove("a", Some("1"));
+    /// assert_eq!(params.get("a"), None);
+    /// ```
+    pub fn remove(&self, key: &str, value: Option<&str>) {
+        if let Some(value) = value {
+            unsafe {
+                ffi::ada_search_params_remove_value(
+                    self.0,
+                    key.as_ptr().cast(),
+                    key.len(),
+                    value.as_ptr().cast(),
+                    value.len(),
+                )
+            }
+        } else {
+            unsafe { ffi::ada_search_params_remove(self.0, key.as_ptr().cast(), key.len()) }
+        }
+    }
+
+    /// Retruns true if the URLSearchParams struct contains the key.
+    ///
+    /// ```
+    /// use ada_url::URLSearchParams;
+    /// let params = URLSearchParams::parse("a=1&b=2");
+    /// assert_eq!(params.has("a", None), true);
+    /// ```
+    pub fn has(&self, key: &str, value: Option<&str>) -> bool {
```

Comment:
```suggestion
    /// Returns whether the [`URLSearchParams`] contains the `key`.
    ///
    /// ```
    /// use ada_url::URLSearchParams;
    /// let params = URLSearchParams::parse("a=1&b=2");
    /// assert_eq!(params.has("a", None), true);
    /// ```
    pub fn contains(&self, key: &str, value: Option<&str>) -> bool {
```

Usually called `contains` in Rust. Same applies as above, maybe making this two separate methods would make sense.

Reply:
Nice. Thanks!